docs/user/spark/core.rst

```diff
@@ -0,0 +1,210 @@
+ * ``geomesa.sft`` - the ``SimpleFeatureType``, as a spec string, configuration string, or environment lookup name
+ * ``geomesa.sft.name`` - (optional) the name of the ``SimpleFeatureType``
+
+Consider the example data described in the :ref:`convert_example_usage` section of the
+:doc:`/user/convert` documentation. If the file ``example.csv`` contains the
+example data, and ``example.conf`` contains the Typesafe configuration file for the
+converter, the following Scala code can be used to load this data into an ``RDD``:
+
+.. code-block:: scala
+
+    val exampleConf = ConfigFactory.load("example.conf").root().render()
+    val params = Map(
+      "geomesa.converter" -> exampleConf,
+      "geomesa.converter.inputs" -> "example.csv",
+      "geomesa.sft" -> "phrase:String,dtg:Date,geom:Point:srid=4326",
+      "geomesa.sft.name" -> "example")
+    val query = new Query("example")
+    val rdd = GeoMesaSpark(params).rdd(new Configuration(), sc, params, query)
+
+.. warning::
+
+    ``ConvertSpatialRDDProvider`` is read-only, and does not support writing features
+    to data files.
+
+.. _geotools_rdd_provider:
+
+GeoTools RDD Provider
+^^^^^^^^^^^^^^^^^^^^^
+
+``GeoToolsSpatialRDDProvider`` is provided by the ``geomesa-spark-geotools`` module:
+
+.. code-block:: xml
+
+    <dependency>
+      <groupId>org.locationtech.geomesa</groupId>
+      <artifactId>geomesa-spark-geotools_2.11</artifactId>
+      // version, etc.
+    </dependency>
+
+``GeoToolsSpatialRDDProvider`` generates and saves ``RDD``\ s of features stored in
+a generic GeoTools ``DataStore``. The configuration parameters passed are the same as
+those passed to ``DataStoreFinder.getDataStore()`` to create the data store of interest,
+plus a required boolean parameter called "geotools" to indicate to the SPI to load
+``GeoToolsSpatialRDDProvider``. For example, the `CSVDataStore`_ described in the
+`GeoTools ContentDataStore tutorial`_ takes a single parameter called "file". To use
+this data store with GeoMesa Spark, do the following:
+
+.. code-block:: scala
+
+    val params = Map(
+      "geotools" -> "true",
+      "file" -> "locations.csv")
+    val query = new Query("locations")
+    val rdd = GeoMesaSpark(params).rdd(new Configuration(), sc, params, query)
+
+.. _GeoTools ContentDataStore tutorial: http://docs.geotools.org/latest/userguide/tutorial/datastore/index.html
+
+.. _CSVDataStore: http://docs.geotools.org/latest/userguide/tutorial/datastore/read.html
+
+The name of the feature type to access in the data store is passed as the type name of the
+query passed to the ``rdd()`` method. In the example of the `CSVDataStore`_, this is the
+basename of the filename passed as an argument.
+
+.. warning::
+
+    Do not use the GeoTools RDD provider with a GeoMesa Accumulo data store. The
+    :ref:`accumulo_rdd_provider` provides additional optimizations to improve performance
+    between Spark/SparkSQL and GeoMesa Accumulo data stores.
```

Comment:
NB: (Maybe worth noting in the docs...)  The k-v, "geotools" -> "true" is required to kick over to the GT RDD Provider.